### ISSUE TITLE
Prework assignment - Please review my app.

### ISSUE THREAD

**Issue by jerryc-nguyen** [jerryc-nguyen/tip-calculator-on-swift#1]
I just completed prework assignment for IOS on swift course, please review.
/cc @coderschoolreview

**Comment by coderschoolreview** [comment]
Hi Nhan,
Thanks for your submission. It looks great. We are going to review and get back to you by next week.

**Comment by avo1** [comment]
Hi Nhan,

👍 Neat and clean code. Good job.

### Quick Comments:
- Good usage of `NSUserDefaults`. This is one of the methods to pass the data from multiple viewController. We will learn few more ways to do that.
- If you were able to save the tipPercentage, why didn't you save the billAmount? (one of the optional). It should be exactly the same.
- You shouldn't let the user enter invalid number, should you? I can enter bunch of leading zeros.
- It's a good practice to validate user's input. Somethings like this should not be possible.
  <img width="317" alt="screen shot 2016-06-12 at 12 58 36 pm" src="https://cloud.githubusercontent.com/assets/10482065/15989338/252784fa-309e-11e6-8941-23ba48d398dc.png">
- It's great you know how to use the tableView. We'll also cover that in detail in class. You can remove the redundant cell in the end of the table by using this `tableView.tableFooterView = UIView()`.  
- Good job with the `UITapGestureRecognizer` 😱  But you can dismiss the keyboard easily by overriding the `touchesBegan` like this:

``` Swift
  override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
      view.endEditing(true)
  }
```

### Optional Suggestions to Improve
- You can show the keyboard immediately when app starts by using `becomeFirstResponder`
- It's OK to use `String(format: "$%.2f", tip)` in this case, but if you change the language to Vietnamese for example, we don't use 2 digits after decimal point and the $ sign, right? I recommend to try `NSNumberFormatter` and `CurrencyStyle`. If I change to Vietnamese, it should show `đ 12,000` for example.

I encourage you to try out the optional tasks to increase your chance. Fyi the deadline was extended to June 24.

Regards,
Dave

**Comment by jerryc-nguyen** [comment]
yes, many thanks for your comments and suggestion to improve the assignment. It's great. I intent to apply to the course which starts at 1st Jun also submit the assignment before knowing about your provided Readme template which include the optional features. So, I will update then submit again and ping you when done. Thanks for your review and support.

**Comment by jerryc-nguyen** [comment]
hi @coderschoolreview , @avo1 I just updated app and need your review again. I completed the optional features also apply your suggest improvements. Looking forward to hearing from you soon.
Thanks.

**Comment by avo1** [comment]
:100: Great edits!! We'll email you about the class soon.

**Comment by jerryc-nguyen** [comment]
yes, many thanks.
